lib/Parse/ParseStmt.cpp

```diff
@@ -2400,3 +2405,54 @@ ParserResult<CaseStmt> Parser::parseStmtCase(bool IsActive) {
                                !BoundDecls.empty(), UnknownAttrLoc, ColonLoc,
                                Body));
 }
+
+/// stmt-pound-assert:
+///   '#assert' '(' expr (',' string_literal)? ')'
+ParserResult<Stmt> Parser::parseStmtPoundAssert() {
+  SyntaxParsingContext AssertContext(SyntaxContext,
+      SyntaxKind::PoundAssertStmt);
+
+  SourceLoc startLoc = consumeToken(tok::pound_assert);
+  SourceLoc endLoc;
+
+  if (!Context.LangOpts.EnableExperimentalStaticAssert) {
+    diagnose(startLoc, diag::pound_assert_disabled);
+    return makeParserError();
+  }
+
+  if (Tok.isNot(tok::l_paren)) {
+    diagnose(Tok, diag::pound_assert_expected_lparen);
+    return makeParserError();
+  }
+  SourceLoc LBLoc = consumeToken(tok::l_paren);
+
+  auto conditionExprResult = parseExpr(diag::pound_assert_expected_expression);
+  // if (conditionExprResult.hasCodeCompletion())
+  //   return conditionExprResult;
+  if (conditionExprResult.isParseError())
+    return makeParserError();
```

Comment:
`return ParserStatus(conditionExprResult)` should be OK.

Reply:
thanks, done!